tests/test_projectgen_extension_optimization_smart1.py

```diff
@@ -0,0 +1,1218 @@
+                }
+                assert set(layer.relevant_topics) == {"Staedtisches_Gewerbe_V1.Firmen"}
+        assert count == 3
+
+        project = Project(
+            optimize_strategy=strategy,
+            context={"catalogue_datasetname": CATALOGUE_DATASETNAME},
+        )
+        project.layers = available_layers
+        project.relations = relations
+        project.legend = legend
+        project.post_generate()
+
+        qgis_project = QgsProject.instance()
+        project.create(None, qgis_project)
+
+        # check layertree
+        root = qgis_project.layerTreeRoot()
+        assert root is not None
+
+        all_layers = root.findLayers()
+        assert len(all_layers) == 7
+
+        geometry_layers = {
+            l.name() for l in root.children() if isinstance(l, QgsLayerTreeLayer)
+        }
+        assert geometry_layers == {"Strasse", "Gebaeude"}
+
+        tables_layers = {l.name() for l in root.findGroup("tables").children()}
+        assert tables_layers == {
+            "Firma",
+            "Gebaeude_StadtFirma",
+            "BesitzerIn",
+        }
+
+        # check relations - all are there
+        relations = list(qgis_project.relationManager().relations().values())
+        assert len(relations) == 10
+
+        # strasse should have relation editors to only one layer (that sums up all the possibilities)
+        count = 0
+        for layer in project.layers:
+            if layer.layer.name() == "Strasse":
+                efc = layer.layer.editFormConfig()
+                # one general and four relation editors
+                assert len(efc.tabs()) == 2
+                for tab in efc.tabs():
+                    if tab.name() == "gebaeude":
+                        count += 1
+                        assert len(tab.children()) == 1
+        # should find 1
+        assert count == 1
+
+        # strasse can only be in it's dedicated basket (only instance of topic Infrastruktur_V1.Strassen)
+        # besitzerin can be in multiple baskets (instances of topics Ortsplanung_V1_1.Konstruktionen, Kantonale_Ortsplanung_V1_1.Konstruktionen, Staedtische_Ortsplanung_V1_1.Freizeit, Staedtische_Ortsplanung_V1_1.Gewerbe)
+        count = 0
+        for layer in project.layers:
+            if layer.layer.name() == "Strasse":
+                count += 1
+
+                # check layer variable
+                layer_model_topic_names = (
+                    QgsExpressionContextUtils.layerScope(layer.layer).variable(
+                        "interlis_topic"
+                    )
+                    or ""
+                )
+
+                assert layer_model_topic_names == "Infrastruktur_V1.Strassen"
+
+                # have look at the basket field
+                fields = layer.layer.fields()
+                field_idx = (
+                    fields.lookupField("T_basket")
+                    if not_pg
+                    else fields.lookupField("t_basket")
```

Comment:
It would be great to have a variable holding the correct basket field name regardless of the engine (e.g., `fields.lookupField(BASKET_FIELD_NAME)`).

I'd have to refresh a bit how the Model Baker code has evolved to remember if this could be added as a class member. Just creating a note here in case you can think of something evident.

Reply:
Thanks I will concern that.